src/kv1/read.rs

```diff
@@ -767,21 +791,89 @@ where
     )?;
 
     make_routes(collections, &list_jopas)?;
+    read_ntcassgn(file_handler, collections, &map_pujopas)?;
 
     // need routes + stop_areas
     // collections.lines = CollectionWithId::new(lines)?;
 
     Ok(())
 }
 
-/// Generates comments on trips
-pub fn read_notice_ntcassgn<H>(file_handler: &mut H, collections: &mut Collections) -> Result<()>
+pub fn read_notice<H>(file_handler: &mut H, collections: &mut Collections) -> Result<()>
 where
     for<'a> &'a mut H: FileHandler,
 {
-    info!("Reading NOTICEXXXX.TMI and NTCASSGNMX.TMI");
+    let file = "NOTICEXXXX.TMI";
+    let (reader, path) = file_handler.get_file(file)?;
+    info!("Reading {}", file);
+
+    let mut rdr = csv::ReaderBuilder::new()
+        .delimiter(b'|')
+        .trim(csv::Trim::All)
+        .from_reader(reader);
+
+    for notice in rdr.deserialize() {
+        let notice: Notice = notice.with_context(ctx_from_path!(path))?;
+        collections
+            .comments
+            .push(Comment {
+                id: notice.notice_code,
+                comment_type: CommentType::default(),
+                label: None,
+                name: notice.notice_content,
+                url: None,
+            })
+            .unwrap();
+    }
 
-    // collections.comments = CollectionWithId::new(comments)?;
+    Ok(())
+}
+
+fn read_ntcassgn<H>(
+    file_handler: &mut H,
+    collections: &mut Collections,
+    map_pujopass: &PujoJopaMap,
+) -> Result<()>
+where
+    for<'a> &'a mut H: FileHandler,
+{
+    let file = "NTCASSGNMX.TMI";
+    let (reader, path) = file_handler.get_file(file)?;
+    info!("Reading {}", file);
+
+    let mut rdr = csv::ReaderBuilder::new()
+        .delimiter(b'|')
+        .trim(csv::Trim::All)
+        .from_reader(reader);
+
+    for notice_assignment in rdr.deserialize() {
+        let notice_assignment: NoticeAssignment =
+            notice_assignment.with_context(ctx_from_path!(path))?;
+
+        if let Some(notice_idx) = collections.comments.get_idx(&notice_assignment.notice_code) {
```

Comment:
```rust
for notice_assignment in rdr.deserialize() {
        let notice_assignment: NoticeAssignment =
            notice_assignment.with_context(ctx_from_path!(path))?;

        if let Some(comment_idx) = collections.comments.get_idx(&notice_assignment.notice_code) {
            if let Some(pujopasses) = map_pujopass.get(&(
                notice_assignment.line_planning_number,
                notice_assignment.journey_number,
            )) {
                for pujopass in pujopasses.iter().filter(|p| p.stop_order == 1) {
                    collections
                        .vehicle_journeys
                        .get_mut(&pujopass.vehiclejourney_id())
                        .unwrap()
                        .comment_links
                        .insert(comment_idx);
                }
            }
        }
    }
```

Reply:
You should be able to replace the `.unwrap()` with something like
```
.ok_or_else(
    format_err!(
        "Failed to find a vehicle journey with ID {}",
        pujopass.vehiclejourney_id()))?
```

Reply:
I wonder if it's really relevant (in logs) because it's not critical. We'll see that tomorrow.

Reply:
I'm pretty sure that this unwrap never fails
and I think you can remove the filter comment_links is a BtreeSet (unique values)